src/main/java/edu/harvard/dbmi/avillach/dictionaryetl/columnmeta/AbstractColumnMetaProcessor.java

```diff
@@ -0,0 +1,297 @@
+package edu.harvard.dbmi.avillach.dictionaryetl.columnmeta;
+
+import com.opencsv.CSVWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+@Component
+public abstract class AbstractColumnMetaProcessor {
+    private final Logger log = LoggerFactory.getLogger(AbstractColumnMetaProcessor.class);
+
+    private final LinkedBlockingQueue<List<ColumnMeta>> readyToLoadMetadata = new LinkedBlockingQueue<>();
+    private final ColumnMetaMapper columnMetaMapper;
+    /**
+     * If a List of ColumnMetas in the readyToLoadMetadata cannot be processed due to an error it will be inserted into
+     * this list.
+     */
+    private final ConcurrentSkipListSet<List<ColumnMeta>> columnMetaErrors =
+            new ConcurrentSkipListSet<>(Comparator.comparing(metas -> metas.getFirst().name())
+            );
```

Comment:
did the formatter do this hanging `);`? that's rude if yes.

Reply:
Maybeeee... maybe not. I am not sure. It was already formatted like this when I copied it over from the `DictionaryLoaderService`. However, this reminds me that we need to add our standard java formatter to the repository.